### DISCUSSION TITLE
GCToOSInterface::QueryPerformanceCounter() in linux

### DISCUSSION THREAD

**Discussion by HJLeee** [dotnet/runtime#75585] [Q&A] [answered]
Hello,

I was wondering if `GCToOSInterface::QueryPerformanceCounter()` in gcenv.unix.cpp is used for linux.
https://github.com/dotnet/runtime/blob/7993e515586bb644d85f6f3b408900dc065f96fd/src/coreclr/gc/unix/gcenv.unix.cpp#L1411-L1425

It is surely compiled but setting a breakpoint to `GCToOSInterface::QueryPerformanceCounter()` points one in gcenv.os.cpp only.
```cpp
(gdb) info b
Num     Type           Disp Enb Address            What
1       breakpoint     keep y   0x00007ffff7076a95 in GCToOSInterface::QueryPerformanceCounter() at /home/leee/dotnet/runtime/src/coreclr/vm/gcenv.os.cpp:1019
        breakpoint already hit 2 times
```

I'd appreciate it If you tell me why the same function exists.

**Comment by jkotas** [comment, marked as answer]
The GC linked into libcoreclr.so uses the implementation from gcenv.os.cpp.

The implementation in gcenv.unix.cpp is used for standalone GC library and for native AOT.

The plan is to delete gcenv.os.cpp and use gcenv.unix.cpp everywhere: https://github.com/dotnet/runtime/issues/72684
